flips/20220811-fungible-tokens-metadata.md

```diff
@@ -0,0 +1,210 @@
+# Fungible Tokens Metadata
+
+| Status        | Draft       |
+:-------------- |:---------------------------------------------------- |
+| **FLIP #1087**    | [1087](https://github.com/onflow/flow/pull/1087)|
+| **Author** | Álvaro Lillo Igualada (alvaro.lillo@dapperlabs.com) |
+| **Sponsor**   | Josh Hannan (joshua.hannan@dapperlabs.com)             |
+| **Updated**   | 2022-08-16                                           | 
+
+## Objective
+
+This proposal, in a similar fashion as the NFT Metadata, will make possible to create generic solutions that will interoperate through views rather than through hard-coded types. This will allow better interoperability between fungible tokens and external applications. 
+
+## Motivation
+
+As the Flow ecosystem continues growing the need of assuring contract and external apps interoperability increases. The development of the NFT Metadata standards brought awareness of how important is the way any token on Flow is able to communicate its features to other dapps, and of the advantages of doing so in a standardized way.
+
+## User Benefit
+
+The creation of a Metadata standard for fungible tokens will bring two major user benefits, discoverability and interoperability. It will make way easier to any FT to be presented on any app that would want to feature several FT and also will ease how those FT can be used by an user programmatically.
+
+## Design Proposal
+
+The core of this proposal is to add the following interface and have any `FungibleToken.Vault` implement it. This is the same interface that can be found in the NFTs metadata contract.
+
+```cadence
+    /// Provides access to a set of metadata views. A struct or 
+    /// resource (e.g. a Vault) can implement this interface to provide access to 
+    /// the views that it supports.
+    ///
+    pub resource interface Resolver {
+        pub fun getViews(): [Type]
+        pub fun resolveView(_ view: Type): AnyStruct?
+    }
+```
+
+Also, based on the experience acquired developing the NFT views a `FTView` that gives a fill picture of the FT, wrapping other views, is considered necessary
```

Comment:
`getViews` would be required to return `FTView` in order to be compatible with the future FT Catalog, just highly recommendable for any token to return that.

In my first draft  the `FTView` looked like this, I don't know why I got rid of the optional types, they are back on the proposal, sorry about that! I believe this way it makes more sense to encourage the wrapper view.

Anyway, `getViews` should also return `FTDisplay` and `FTVaultData` in case some application does not care about the other view. I should make that more clear on the proposal.

And also I'm totally fine removing the wrapper view, I was just copycatting the final solution on the NFT metadata.

```   
pub struct FTView {
        pub let ftDisplay: FTDisplay?
        pub let vaultData: FTVaultData?
        init(
            ftDisplay : FTDisplay?,
            vaultData : FTVaultData?
        ) {
            self.ftDisplay = ftDisplay
            self.vaultData = vaultData
        }
}
```

Do you think this should be something that must be settled before getting the flip into the forum or that is worth of being discussed there?

Reply:
I think the view is still worth including, but it should also include some of the views from the NFT `MetadataViews` contract because it is meant to represent a single place to get all the important views that can be implemented by an FT and it is nice having it in one place. So I would include those here and also in your PR when you open that in the flow-ft repo

Reply:
I could see maybe wanting to do `Royalties` and `License`, but I think the ones you mention are the main ones. I don't know if they all belong in the `FTVaultDisplay` view though. They could probably just be implemented separately in the example contract.

So what I am saying is:
1. What you have here looks good. When you write this contract in the flow-ft repo, do the views as they are shown here.
2. In the `ExampleToken` contract, implement the views that you have here and also include resolvers for `Media` and `License`. Don't include one for royalties though because that probably doesn't belong in the example contract.

Reply:
great, that makes a lot of sense, thank you!!